query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala

```diff
@@ -431,7 +439,23 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
     }
 
     analysisMap collect {
-      case (k, Success(a)) => k -> a
+      case (k, Success(a)) =>
+        val ruAnalysis =
+          if (project.nonEmpty) {
+            val reprojectedSelection = a.selection.map {
+              case x@(cn: ColumnName, expr) =>
+                project.get(cn.name.stripMargin('_')) match {
```

Comment:
Doing things based on the contents of a function of the name seems odd to me.  What's the reason for it?